CMSgov/hcgov-design-system
# [WNMGDS-991] add husky and lint-staged

## Summary
Using Husky and lint-staged to lint all staged files before committing

## Added 
Added Husky and lint staged 

## How to test
Make a linting issue in a JSX, TSX or SCSS file and stage it for commit and ensure that it gets either autofixed or flagged for manual fix before allowing the commit
